libs/ballot-interpreter/src/bmd/interpret.test.tsx

```diff
@@ -0,0 +1,183 @@
+import { assert, assertDefined, err, iter, typedAs } from '@votingworks/basics';
+import {
+  electionFamousNames2021Fixtures,
+  electionGridLayoutNewHampshireTestBallotFixtures,
+  sampleBallotImages,
+} from '@votingworks/fixtures';
+import { pdfToImages } from '@votingworks/image-utils';
+import { renderToPdf } from '@votingworks/printing';
+import {
+  ElectionDefinition,
+  SheetOf,
+  VotesDict,
+  vote,
+} from '@votingworks/types';
+import { BmdPaperBallot } from '@votingworks/ui';
+import { Buffer } from 'buffer';
+import { ImageData } from 'canvas';
+import { InterpretError, interpret } from './interpret';
+
+async function renderTestModeBallot(
+  electionDefinition: ElectionDefinition,
+  precinctId: string,
+  ballotStyleId: string,
+  votes: VotesDict
+): Promise<Buffer> {
+  const ballot = (
+    <BmdPaperBallot
+      electionDefinition={electionDefinition}
+      ballotStyleId={ballotStyleId}
+      precinctId={precinctId}
+      votes={votes}
+      isLiveMode={false}
+      generateBallotId={() => '1'}
+      machineType="mark"
+    />
+  );
+
+  return renderToPdf({
+    document: ballot,
+  });
+}
+
+async function convertBallotToImages(pdf: Buffer): Promise<SheetOf<ImageData>> {
+  const [page1, page2] = await iter(pdfToImages(pdf, { scale: 200 / 72 }))
+    .map((page) => page.page)
+    .take(2)
+    .toArray();
+
+  return [
+    assertDefined(page1),
+    // if there's only one page, use a blank page for the back
+    page2 ??
+      (await electionFamousNames2021Fixtures.machineMarkedBallotPage2.asImageData()),
+  ];
+}
+
+const famousNamesVotes = vote(
+  electionFamousNames2021Fixtures.election.contests,
+  {
+    mayor: 'sherlock-holmes',
+    controller: 'winston-churchill',
+    attorney: 'john-snow',
+    'public-works-director': 'benjamin-franklin',
+    'chief-of-police': 'natalie-portman',
+    'parks-and-recreation-director': 'charles-darwin',
+    'board-of-alderman': [
+      'helen-keller',
+      'steve-jobs',
+      'nikola-tesla',
+      'vincent-van-gogh',
+    ],
+    'city-council': [
+      'marie-curie',
+      'indiana-jones',
+      'mona-lisa',
+      'jackie-chan',
+    ],
+  }
+);
+
+let famousNamesBmdCard: SheetOf<ImageData>;
+
+beforeAll(async () => {
+  famousNamesBmdCard = await convertBallotToImages(
+    await renderTestModeBallot(
+      electionFamousNames2021Fixtures.electionDefinition,
+      '23',
+      '1',
+      famousNamesVotes
+    )
+  );
+});
+
+test('happy path: front, back', async () => {
+  const card = famousNamesBmdCard;
+  const result = await interpret(
+    electionFamousNames2021Fixtures.electionDefinition,
+    card
+  );
+  const { ballot, summaryBallotImage, blankPageImage } = result.unsafeUnwrap();
+  expect(ballot).toMatchSnapshot();
+
+  // don't use Jest `toEqual` matcher because it tries to pretty print the
+  // ImageData objects, which is slow and causes the test to time out
+  assert(summaryBallotImage === card[0]);
+  assert(blankPageImage === card[1]);
```

Comment:
I've used `jest-image-matcher` for image comparison, though if this is just for object identity then maybe you don't need it

Reply:
As there doesn't appear to be anything set up to compare two images instead of one image vs a snapshot, I'm going to leave this as is for now.

Reply:
Here is an example: https://github.com/votingworks/vxsuite/blob/main/libs/hmpb/src/generate_fixtures.test.ts#L30-L33

Reply:
But I agree that it's fine to leave as is